src/transformers/configuration_utils.py

```diff
@@ -1122,6 +1122,16 @@ def get_text_config(self, decoder=False) -> "PretrainedConfig":
             )
         elif len(valid_text_config_names) == 1:
             return getattr(self, valid_text_config_names[0])
+        else:
+            # In case no valid text config is found, we might have a model with a vlm backbone
+            if hasattr(self, "vlm_config"):
+                for text_config_name in possible_text_config_names:
+                    if hasattr(self.vlm_config, text_config_name):
+                        text_config = getattr(self.vlm_config, text_config_name, None)
+                        if text_config is not None:
+                            valid_text_config_names += [text_config_name]
+            if len(valid_text_config_names) == 1:
+                return getattr(self.vlm_config, valid_text_config_names[0])
```

Comment:
This should not be needed, if your config is a compostion config, see my suggestion

Reply:
@yonigozlan What do you think? Since you were the one who added it, I'll let you decide on whether it's necessary or not given Arthur's suggestion!

Reply:
@ArthurZucker Without this, even by setting `vlm_config` as a sub_config, `PretrainedConfig` can't find the `text_config`.
The difference here is that we have nested sub_configs, where the text_config we want to get is itself a sub config of `vlm_config`. I don't think we have nested composition configs in other Transformers models so this would be a first, but since ColPali is essentially a wrapper around a VLM, I'd say it would make sense to introduce this here, wdyt?

Reply:
I think it makes more sense to flatten whatever we can rather than introducing this! 🤗 You don't even need a new config class for ColPaliGemma, you can use the PaliGemma one!

Reply:
A big advantage of having the `vlm_config` is that it allows to swap the vlm backbone easily, which is how `ColQwen` or `ColSmol` work I think cc @tonywu71 . If we just use the PaliGemma config, users won't be able to experiment with other backbones. Another possibility would be to add a `text_config` at the "root" of the ColPali config, that would overwrite the `text_config` inside the `vlm_config` if provided, so that we wouldn't need to have this `else` here

Reply:
Agreed! Personally, I think it'd be great to give the community the freedom of experimenting with the ColPali architecture freely with other VLM backbones. Wdyt?

Reply:
Sounds good! I removed the changes in `configuration_utils.py` and added a `text_config` arg to `ColPaliConfig` which overrides the `text_config` of the vlm if provided. Should be good now!